src/openqaoa-core/openqaoa/algorithms/baseworkflow.py

```diff
@@ -373,6 +373,12 @@ def compile(self, problem: QUBO):
         assert isinstance(problem, QUBO), "The problem must be converted into QUBO form"
         self.problem = problem
 
+        if hasattr(self.device, "n_qubits"):
+            if self.device.n_qubits < self.problem.n:
+                raise Exception(
+                    "There are lesser qubits on the device than the number of qubits required for the circuit."
```

Comment:
I always find it useful seeing the offending numbers :D 

"The number of qubits {n_qubit} is lesser than the number of qubits available on the device: f{self.device.name} features f{self.device.n_qubits} qubits"

Reply:
done